### PR TITLE
Improved project setup, added some MC integration 

### PR DESCRIPTION
**Pull request by xtphty**

### CODE REVIEW

**Comment by hayksaakian** [comment]
As i understand it: Private key is basically a key held on the minecraft server, authtoken is a users destiny gg api key. 

If you send both with a uuid you can know that an exact minecraft user is requesting permission to a valid minecraft server, and with the apikey you can validate their sub level.

**Comment by sztanpet** [comment]
correct, the private key is purely to protect the dgg api from "abuse"

**Comment by sztanpet** [comment]
also, to give you a hint about using the api, ask me in private for the privatekey and authtokens for users with and without a sub

```
THIS IS A GET:
curl -v "stage.destiny.gg/auth/minecraft?privatekey=ccccccccccccccccccc&uuid=bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
THIS IS A POST:
curl -v --data "privatekey=ccccccccccccccccccc&authtoken=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa&uuid=bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb" stage.destiny.gg/auth/minecraft
```

File: src/main/java/net/bitjump/bukkit/subwhitelister/commands/RegisterCommand.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.java.JavaPlugin;
+
+/**
+ * Registers a user with their privateKey
+ * @author xtphty
+ *
+ */
+public class RegisterCommand implements CommandExecutor {
+	private static final String USAGE = "Invalid command, use: /register <private key>";
```
Comment (sztanpet) [inline src/main/java/net/bitjump/bukkit/subwhitelister/commands/RegisterCommand.java]: this is what we are talking about by the way, it should say `<login key from destiny.gg>` by default and make it configurable from the config
the private key is also a configurable, never exposed thing that is simply sent along with everything to prove that the minecraft server is the one doing the api request
so the authToken is what we are asking the user for, and it is exposed as "login key" in the d.gg profile

**Comment by xtphty** [comment]
Can I get admin on stage.dgg? want to test this out tomorrow using a local server will be nice if i can give / take sub away from people.

**Comment by sztanpet** [comment]
no such feature

**Pull request merged**